### ISSUE TITLE
firestore on nodejs server is very slow

### ISSUE THREAD

**Issue by zakaria-chahboun** [firebase/firebase-admin-node#748]
i'm using Firestore (8.9.0) with Nodejs (v12.13.0) (expressjs framework) and the first simple request takes more than 30 seconds!

and if i get the same request with JavaScript in the client-side or with Rest API tool like (postman) i get the data immediately! no problem!
i also try the same example on python and no problem with it!

So what is the issue here about firestore package on nodejs?

this is the code:

```javascript
var express = require('express');
var router = express.Router();
    
    
    const admin = require("firebase-admin");
    
    const serviceAccount = require("../privte/quiztestweb-firebase-adminsdk-vd7zn-1a3a896f60.json");
    
    admin.initializeApp({
      credential: admin.credential.cert(serviceAccount),
      databaseURL: "https://quiztestweb.firebaseio.com"
    });

    const db = admin.firestore();

    /* GET users listing. */
    router.get('/', function (req, res, next) {

      db.collection("/quizzes/quiz_1/tests/").get()
        .then((snapshot) => {
          snapshot.forEach((doc) => {
            console.log(doc.id, '=>', doc.data());
            res.send("it's works!");
          });
        })
        .catch((err) => {
          console.log('Error getting documents', err);
          res.send('BAD THING!');
        });
    
    });
    
    module.exports = router;
```

**Comment by zakaria-chahboun** [comment]
**Finally!**

i solve this issue by removing the new version of firebase package and replace it with an older one!

**_((but it's still just a temporary solution .. until fixed by firebase developers))_**

**Comment by hiranya911** [comment]
I tried your code on Node 8  and firebase-admin@8.9.0 where it seems to work fine. I get the response immediately. Your code also has a bug, where it will call `res.send()` multiple times for the same request, due to it being called in a `forEach` block.

**Comment by hiranya911** [comment]
Just tried it on Node 12, where it works fine too. My collection has around 7 documents. Here's my test code (I've fixed the bug in your original code) and  the curl output when calling it:

```
var express = require('express');
var router = express.Router();
var app = express();

const admin = require("firebase-admin");

const serviceAccount = require("~/path/to/serviceAccount.json");

admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
});

const db = admin.firestore();
router.get('/', function (req, res, next) {
    db.collection("movie_reviews").get()
      .then((snapshot) => {
        snapshot.forEach((doc) => {
          console.log(doc.id, '=>', doc.data());
        });
        res.send("it's works!");
      })
      .catch((err) => {
        console.log('Error getting documents', err);
        res.send('BAD THING!');
      });

  });

app.use('/', router);
app.listen(8080, () => console.log(`Example app listening on port 8080!`))
```

Output (first time invocation):
```
$ time curl -v localhost:8080
* Rebuilt URL to: localhost:8080/
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< X-Powered-By: Express
< Content-Type: text/html; charset=utf-8
< Content-Length: 11
< ETag: W/"b-/gRBeh/mWxH50B7f/DSBFOW8XqQ"
< Date: Sat, 04 Jan 2020 21:27:41 GMT
< Connection: keep-alive
< 
* Connection #0 to host localhost left intact
it's works!
real	0m0.637s
user	0m0.001s
sys	0m0.008s
```

So around 600ms for total roundtrip.

**Comment by hiranya911** [comment]
@zakaria-chahboun can you try the above experiment and report back your observations. If you can reproduce the issue consistently, please share a Firestore log with us. Add the following to your code to enable logging:

```
admin.firestore.setLogFunction(console.log);
```

**Comment by schmidt-sebastian** [comment]
@zakaria-chahboun Can you also let us know which two versions of `@google-cloud/firestore` you are comparing? You can get the underlying version via `npm list`.

**Comment by schmidt-sebastian** [comment]
We have received logs in https://github.com/firebase/firebase-admin-node/issues/750. I am currently looking into this issue.

**Comment by schmidt-sebastian** [comment]
This has been fixed and released as Firestore 3.3.1.

**Comment by zakaria-chahboun** [comment]
> @zakaria-chahboun Can you also let us know which two versions of `@google-cloud/firestore` you are comparing? You can get the underlying version via `npm list`.

7.1.0 and 8.9.0
